### PR TITLE
Add judicialmind/greenleaf-law-embed-tiny

### PR DESCRIPTION
**Pull request by Surya-saka**
## Model: judicialmind/greenleaf-law-embed-tiny

596M-parameter dense embedding model purpose-built for legal text retrieval.

### ModelMeta Details
- **Name:** judicialmind/greenleaf-law-embed-tiny
- **Parameters:** 595,776,512 (596M)
- **Embedding dim:** 1024
- **Max tokens:** 32,768
- **License:** Apache 2.0
- **Open weights:** Yes
- **Framework:** Sentence Transformers, PyTorch, safetensors
- **Similarity:** Cosine
- **Revision:** 7939f1bf2945e06f75191daa80feee191af38d13

### Architecture
Bidirectional transformer encoder derived from Qwen3 architecture. Causal masking removed across all 28 layers for full-document understanding. Mean pooling. Built-in int8/binary quantization.

### Custom Loader
Uses `GreenLeafEmbedWrapper` (extends `SentenceTransformerEncoderWrapper`) which loads with `trust_remote_code=True` to support the custom bidirectional attention and quantization modules.

### Benchmark Results (MTEB(Law, v1))

| Task | Score (nDCG@10) |
|------|-----------------|
| AILACasedocs | 40.91 |
| AILAStatutes | 58.79 |
| GerDaLIRSmall | 38.50 |
| LeCaRDv2 | 69.77 |
| LegalBenchConsumerContractsQA | 85.89 |
| LegalBenchCorporateLobbying | 95.05 |
| LegalQuAD | 58.84 |
| LegalSummarization | 68.19 |
| **MEAN** | **64.49** |

### Checklist
- [x] I have filled out the ModelMeta object to the extent possible
- [x] I have ensured that my model can be loaded using `mteb.get_model(model_name, revision)` and `mteb.get_model_meta(model_name, revision)`
- [x] I have tested the implementation works on a representative set of tasks (all 8 MTEB(Law, v1) tasks)
- [x] The model is public — weights are publicly available on HuggingFace
- [x] Training data is proprietary (not disclosed) — same approach as Voyage/OpenAI submissions

### CODE REVIEW

File: mteb/models/model_implementations/greenleaf_models.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def __init__(self, model_name: str, revision: str | None = None, **kwargs):
+        from sentence_transformers import SentenceTransformer
+
+        model = SentenceTransformer(
+            model_name,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        super().__init__(model=model, **kwargs)
+
```
Comment (ayush1298) [inline mteb/models/model_implementations/greenleaf_models.py]: no need of these, you can directly use `SentenceTransformerEncoderWrapper`

**Comment by Surya-saka** [comment]
Addressed the review: removed custom `GreenLeafEmbedWrapper`, now using `SentenceTransformerEncoderWrapper` directly with `loader_kwargs={"trust_remote_code": True}`.

**Comment by ayush1298** [comment]
LGTM !
You can remove `mteb_mock_run_results.md`
@Samoed can you check?

**Comment by Surya-saka** [comment]
Done — removed mteb_mock_run_results.md

File: mteb/models/model_implementations/greenleaf_models.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    max_tokens=32768,
+    embed_dim=1024,
+    license="apache-2.0",
+    similarity_fn_name=ScoringFunction.COSINE,
+    framework=["Sentence Transformers", "PyTorch", "safetensors"],
+    use_instructions=False,
+    reference="https://huggingface.co/judicialmind/greenleaf-law-embed-tiny",
+    public_training_code=None,
+    public_training_data=None,
+    training_datasets=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/greenleaf_models.py:34]: Can you fill you training datasets?

File: mteb/models/model_implementations/greenleaf_models.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"""
+
+from __future__ import annotations
+
+import logging
+
+from mteb.models.model_meta import ModelMeta, ScoringFunction
+from mteb.models.sentence_transformer_wrapper import SentenceTransformerEncoderWrapper
+
+logger = logging.getLogger(__name__)
```
Comment (Samoed) [inline mteb/models/model_implementations/greenleaf_models.py]:
```suggestion
```

**Comment by Surya-saka** [comment]
Added public_training_data pointing to https://huggingface.co/datasets/judicialmind/legal-training-dataset — our open training corpus with 3.4M+ pairs including 150K+ hand-curated legal query-passage pairs across multiple jurisdictions and document types.
